File: AI-Career-Coach-frontend/src/modules/mock-interview-v2/types/interviewer.ts
export interface InterviewerVoiceV2 {
  id: string;
  name: string;
  gender: "female" | "male";
  accent: string;
  provider: "deepgram" | "browser";
}

export interface InterviewerProfileV2 {
  id: "female_hr" | "female_tech" | "male_hr" | "male_tech" | string;
  name: string;
  role: string;
  gender: "female" | "male";
  experience: string;
  specialization: string;
  personality: string;
  interview_style: string;
  default_voice_id: string;
  voice_preference: string;
  office_setting: string;
  description: string;
  avatar_video_src: string;
  avatar_thumbnail_src: string;
  background_backdrop_src: string;
  is_photorealistic: boolean;
  voices: InterviewerVoiceV2[];
}

File: AI-Career-Coach-Backend/app/mock_interview_v2/schemas/interviewer.py
from typing import Dict, Any, List, Optional
from pydantic import BaseModel, Field


class InterviewerVoiceSchema(BaseModel):
    id: str
    name: str
    gender: str
    accent: str
    provider: str = "deepgram"


class InterviewerProfileSchema(BaseModel):
    id: str
    name: str
    role: str
    gender: str
    experience: str
    specialization: str
    personality: str
    interview_style: str
    default_voice_id: str
    voice_preference: str = "en-US"
    office_setting: str
    description: str
    avatar_video_src: str
    avatar_thumbnail_src: str
    background_backdrop_src: str
    voices: List[InterviewerVoiceSchema] = []
    is_photorealistic: bool = True


class InterviewerCatalogResponse(BaseModel):
    success: bool = True
    interviewers: List[InterviewerProfileSchema]
    total: int
